pages/api/applications/index.js

```diff
@@ -1,8 +1,32 @@
-import { getApplications } from "../../../actions/applications";
+import Application from "../../../models/Application";
+import { getSession } from "next-auth/react";
+import queryWithSession from "../../../utils/queryWithSession";
 
 export default async function handler(req, res) {
   if (req.method === "GET") {
     return res.status(200).send(await getApplications());
+  } else if (req.method === "POST") {
+    const session = await getSession({ req });
+
+    if (session) {
+      const application = req?.body;
+      if (!application) {
+        res.status(400).json({ error: "Malformed request", success: false });
+      } else {
+        const { result, error } = await queryWithSession((session) => {
+          Application.create(application);
```

Comment:
```suggestion
          Application.create([application], { session });
```
We just have to pass the `session` here, a bit cumbersome, I know ... and a side effect of this is that, as per documentation, the first parameter does have to be an array then: https://mongoosejs.com/docs/api/model.html#model_Model.create

Reply:
Why are we passing the session here? I'll re-test locally and recommit.

Reply:
Apparently there is a long standing bug in either MongoDB Core or Mongoose adapter, that you can mitigate by passing the session manually :slightly_smiling_face: 
This was at least what I found out after we were seeing some 500 errors in production and we didn't see them since then...
Original reference:
https://stackoverflow.com/a/69930372

Reply:
Ooo ok. Great, i just pushed the change.